icu4c/source/common/umutex.cpp

```diff
@@ -55,9 +56,11 @@ std::once_flag initFlag;
 std::once_flag *pInitFlag = &initFlag;
 
 }  // Anonymous namespace
+#endif
 
 U_CDECL_BEGIN
 static UBool U_CALLCONV umtx_cleanup() {
+#if U_HAVE_ATOMICS
     initMutex->~mutex();
     initCondition->~condition_variable();
     UMutex::cleanup();
```

Comment:
UMutex::cleanup() is not empty. Should it not still be called? And if so, then umtx_init() should still do `ucln_common_registerCleanup(UCLN_COMMON_MUTEX, umtx_cleanup)`.

Reply:
On second thought, I realized we don't need `umtx_cleanup` and `umtx_init` at all because:
1. `gListHead` is not used, it's always nullptr (thus `UMutex::cleanup` is always no-op)
2. We don't even use actual std::mutex, so we don't need global shared mutex (`initMutex`, `initCondition`, and `globalMutex`)